
<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
 <p style="text-align:right;">
     שבץ (באנגלית: Stroke או Cerebro Vascular Accident ובקיצור CVA) הוא אירוע חירום רפואי, שעלול להביא לפגיעה עצבית קבועה, לסיבוכים ולמוות. הוא הגורם העיקרי לנכות אצל מבוגרים בעולם המערבי והסיבה השנייה למוות ברחבי העולם. בישראל, לפי נתוני משרד הבריאות, מתרחשים כ-15,000 מקרים חדשים של שבץ מוחי מדי שנה. שבץ מתרחש כאשר ישנה הפרעה פתאומית באספקת הדם לרקמת המוח. ההפרעה הזאת גורמת נזק נוירולוגי אשר יכול לבוא לידי ביטוי בדרכים שונות ובהן הפרעה בדיבור, חולשה או שיתוק של הגפיים וקשיים בהליכה. לעיתים עלול אירוע מוחי לגרום למוות. התרחשות של שבץ היא תהליך מהיר שבסופו, בגלל הפגיעה בזרימת הדם למוח, תאי מוח ניזוקים והתפקוד שעליו אחראי האזור הפגוע במוח ניזוק גם הוא. לדוגמה, אם נפגע האזור במוח ששולט על יכולת הדיבור, ייפגעו התפקוד השפתי ומרכז הדיבור של החולה
    </p>
   
    
 </div> 



<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
    <h4>  
את הנתונים כבר ניתחו הרבה מדעני נתונים לכן מה שאציג כאן הוא איך לגשת למידע ולנתח אותו 
 

    
</h4> </div> 

https://www.kaggle.com/fedesoriano/stroke-prediction-dataset/download


 <style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
    נייבא כמה סיפריות שנרחיב עליהן בפוסטים נפרדים 

    
 </div> 
    


In [6]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
כאשר אנחנו מגדירים מחלקה חדשה ב-CSS, עלינו לפתוח סוגריים מסולסלים ולהגדיר את התכונות של כל מחלקה ומחלקה. עבור המחלקה ```rtl``` לדוגמה, הגדרנו שכיוון הכתיבה הוא מימין לשמאל, לכן בחרנו להשתמש בתכונה ```direction``` שמקבלת את הערך ```rtl```.
</div>


<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
נגדיר משתנה src בתור המקור ממנו נייבא את הנתונים ונשתמש בפונקציה בשם read_csv מתוך ספרית Pandas
</div>

In [ ]:
src = r'C:\Users\ayose\GDrive\Data Science\Stroke'
s_data = pd.read_csv(src+'/healthcare-dataset-stroke-data.csv')



    
<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
דבר ראשון נסתכל על הנתונים מלמעלה:
</div>

In [11]:
s_data.head(3)

,id,gender,age,hypertension,heart_disease,ever_married,work_type,Residence_type,avg_glucose_level,bmi,smoking_status,stroke
0,9046,Male,67.0,0,1,Yes,Private,Urban,228.69,36.6,formerly smoked,1
1,51676,Female,61.0,0,0,Yes,Self-employed,Rural,202.21,NaN,never smoked,1
2,31112,Male,80.0,0,1,Yes,Private,Rural,105.92,32.5,never smoked,1


 <div dir="auto">
כאשר אנחנו מגדירים מחלקה חדשה ב-CSS, עלינו לפתוח סוגריים מסולסלים ולהגדיר את התכונות של כל מחלקה ומחלקה. עבור המחלקה ```rtl``` לדוגמה, הגדרנו שכיוון הכתיבה הוא מימין לשמאל, לכן בחרנו להשתמש בתכונה ```direction``` שמקבלת את הערך ```rtl```.
</div>


In [10]:
# ראשית נבדוק כמה נתונים פגומים יש לנו
s_data.isna().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64


<style>
  .rtl { direction:rtl;}
</style>

 <div dir="auto">
ניתן לראות שבערכי ה BMI ישנם 201 ערכים לא מספריים (NaN)

</div>

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64